test/unit/api/local-storage.spec.js

```diff
@@ -274,6 +277,18 @@ describe('LocalStorage', () => {
           done();
         });
       });
+
+      test('should not update if the metadata match', done => {
```

Comment:
I see this test a bit hacky. At the end what you I expect you want to verify is whether
```
this._writePackage(name, packageLocalJson, function(err) {
          callback(err, packageLocalJson);
        });
```
was not called at all, then you can rely the file on storage has not been modified. I'd suggest mock methods whether is possible. (like we started to experiment here https://github.com/verdaccio/local-storage/blob/master/src/___tests___/local-database.test.js)

Aside of that, `test()` blocks should be independent each other, share state each other is not good idea, breaking then would be easier. 

Actually fails on my machine.

```
  ● LocalStorage › LocalStorage::preparePackage › LocalStorage::updateVersions › should not update if the metadata match

    expect(object).toHaveProperty(path, value)

    Expected the object:
      {"atime": 2019-01-29T22:29:28.686Z, "atimeMs": 1548800968685.8875, "birthtime": 2019-01-29T22:29:28.686Z, "birthtimeMs": 1548800968685.8875, "blksize": 4096, "blocks": 8, "ctime": 2019-01-29T22:29:28.687Z, "ctimeMs": 1548800968686.525, "dev": 16777220, "gid": 20, "ino": 8674425728, "mode": 33188, "mtime": 2019-01-29T22:29:28.686Z, "mtimeMs": 1548800968686.1055, "nlink": 1, "rdev": 0, "size": 2723, "uid": 501}
    To have a nested property:
      "ctime"
    With a value of:
      2019-01-29T22:29:28.681Z
    Received:
      2019-01-29T22:29:28.687Z

    Difference:

    - Expected
    + Received

    - 2019-01-29T22:29:28.681Z
    + 2019-01-29T22:29:28.687Z
```
For a few milliseconds. 

So, aside of the patch which I'm 200% agree. But I cannot let this test pass.

Reply:
Thanks @buffaybu changes are quite good :-)